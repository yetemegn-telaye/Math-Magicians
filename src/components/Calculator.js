import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Button from './Button';
import Header from './Header';
import '../styles/calculator.css';

const Calculator = () => {
  const [calcObj, setCalcObj] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const onClickHandler = (event) => {
    const buttonName = event.target.innerHTML;
    const { total, next, operation } = calcObj;

    const res = calculate({ total, next, operation }, buttonName);
    setCalcObj(res);
  };
  const { total, next, operation } = calcObj;
  return (
    <>
      <Header />
      <div className="calc-body">
        <h3>Lets do some math!</h3>
        <div className="calculator-container">
          <div className="grid-item result">
            {total}
            {operation}
            {next}
          </div>
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="AC" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="+/-" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="%" />
          <Button classN="grid-item operator-item" onClickHandler={onClickHandler} innerVal="รท" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="7" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="8" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="9" />
          <Button classN="grid-item operator-item" onClickHandler={onClickHandler} innerVal="x" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="4" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="5" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="6" />
          <Button classN="grid-item operator-item" onClickHandler={onClickHandler} innerVal="-" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="1" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="2" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="3" />
          <Button classN="grid-item operator-item" onClickHandler={onClickHandler} innerVal="+" />
          <Button classN="grid-item zero" onClickHandler={onClickHandler} innerVal="0" />
          <Button classN="grid-item" onClickHandler={onClickHandler} innerVal="." />
          <Button classN="grid-item operator-item" onClickHandler={onClickHandler} innerVal="=" />
        </div>
      </div>

    </>
  );
};

export default Calculator;
