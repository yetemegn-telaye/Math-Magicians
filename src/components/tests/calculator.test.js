import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import Calculator from '../Calculator';

afterEach(() => {
  cleanup();
});

test('Calculator component rendered', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
