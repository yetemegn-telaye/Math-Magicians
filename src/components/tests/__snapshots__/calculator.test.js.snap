// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component rendered 1`] = `
<div
  className="calc-body"
>
  <h3>
    Lets do some math!
  </h3>
  <div
    className="calculator-container"
  >
    <div
      className="grid-item result"
    >
      0
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="grid-item operator-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="grid-item operator-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="grid-item operator-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="grid-item operator-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="grid-item zero"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="grid-item operator-item"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
